library/HTMLPurifier/Length.php

```diff
@@ -78,7 +78,7 @@ protected function validate()
         if ($this->n === '0' && $this->unit === false) {
             return true;
         }
-        if (!ctype_lower($this->unit)) {
+        if ($this->unit === false || !ctype_lower($this->unit)) {
```

Comment:
Shouldn't this be the other way around: if unit is false, then we never need to strtolower the unit

Reply:
I was just trying to keep the behavior the same, but you may be right that this was masking a bug

Reply:
Maybe the clearest thing to do is simply return false if unit is false here: you've already passed the "zero" cases where "no unit" is allowed: the old behavior was turning the false into the empty string via strtolower and then failing later when looking that empty string up in allowedUnits.

If there's no objection to the construction, probably the "cleanest" version of that change leaves this ctype_lower check alone and instead replaces the immediately previous check with:

```
if ($this->unit === false) {
    return ($this->n === '0');
}
```

Reply:
@zerocrates I like your suggestion, its remove the invalid unit and no error for ctype_lower() appears.